src/runtime_src/core/pcie/driver/linux/xocl/devices.h

```diff
@@ -281,6 +280,7 @@ enum subdev_id {
 	XOCL_SUBDEV_FMGR,
 	XOCL_SUBDEV_MIG_HBM,
 	XOCL_SUBDEV_MAILBOX_VERSAL,
+	XOCL_SUBDEV_P2P,
```

Comment:
P2P has to be before MB_SCHEDULER. Could you try to move XOCL_SUBDEV_MAILBOX after XOCL_SUBDEV_FLASH?